css-view-transitions-1/Overview.bs

```diff
@@ -197,38 +197,59 @@ urlPrefix: https://html.spec.whatwg.org/multipage/rendering.html; type: dfn;
 
 # Pseudo-elements # {#pseudo}
 
-	The following <dfn export>view-transition pseudo-elements</dfn> represent the various items being animated.
+## Pseudo-element parents ## {#parent-pseudo}
 
-	The ''::view-transition'' pseudo-element acts as a grouping element for other [=view-transition pseudo-elements=]
-	and has the document's [=document element=] as its [=originating element=].
+	A <dfn>pseudo-element parent</dfn> is a mixin that has <dfn for="pseudo-element parent">children</dfn>,
+	a [=/list=] of [=pseudo-elements=].
+	Initially an empty [=/list=].
 
-	Note: For example, '':root::view-transition'' selector matches this pseudo-element,
-	but ''div::view-transition'' does not.
+	A [=pseudo-element parent=]'s [=pseudo-element parent/children=] are laid out in the order they appear in the [=/list=], according to their styles.
```

Comment:
I'm worried we'll end up re-defining a lot of DOM concepts if we have to define how a tree of pseudo-elements works. For instance, this still leaves hit-testing in an ambiguous state.

Also, implementation wise a pseudo-element tree works exactly like a [node](https://dom.spec.whatwg.org/#node) tree. I mean if you inserted nodes using the script API in the same order (as defined in this spec), style, layout, paint, hit-testing would all work the same as a script generated DOM tree. The only difference here is that the Node APIs are not exposed to the developer.

So could we just clarify the difference between pseudo-element and node, and then lean on same spec concepts as a node tree for this pseudo-element tree?

Reply:
> So could we just clarify the difference between pseudo-element and node

I don't think that should be this spec's job. Presumably the details around how pseudo-elements handle clicks is already defined for things like `::before` and `::after`.

The additions we need compared to `::before` and `::after`:

- A clear tree structure. Although `::before::marker` already has nesting, it's hand-waved in a way that I don't think it suitable for the level of nesting we have.
- Insert and remove. `::before` and `::after` are kinda 'always there', whereas ours come and go.
- Ordering within the parent. `::before` and `::after` [hand-wave this in a kinda ambiguous way](https://w3c.github.io/csswg-drafts/css-pseudo/#generated-content).

My intent here is that [=pseudo-element parent=]'s [=pseudo-element parent/children=] defines that the pseudos are document children of the parent, in that order.

I don't think we need to define hit testing, because (I assume) the hit testing of pseudo-elements is already defined, and it would be redundant to say that these pseudo-elements behave like pseudo-elements.